### PR TITLE
AU-410: Refactor Contact person / Address input.

### PR DESCRIPTION
**Pull request by jiisuominen**
Refactors address / contacts on application form to selectable values. Also bank account / official / address deletion on grants profile.

So:
- Pull / make new
- Go to grants profile and add / delete account, official & address
- Go to form, fill things, see that bank account / address / official are dropdowns
- Finish application (note that review page is not updated with these new changes)
- Go to edit application and see that your selections are still on the form.

This was surprisingly tedious task but things should work. Be vigilante for errors as there must be some.

### CODE REVIEW

**Review by strangeworlder — COMMENTED** [review]
Lomakesivuilla tulee seuraavat errorit:

There was an unexpected problem serving your request. Please try again and contact us if the problem persist including e505f3b7-5576-430f-b93d-be61218822c5 in your message.

Warning: Undefined array key "#default_value" in grants_handler_webform_element_alter() (line 244 of modules/custom/grants_handler/grants_handler.module).

There was an unexpected problem serving your request. Please try again and contact us if the problem persist including 9e012638-5145-4b49-8917-8fb1cd6bdf44 in your message.

Warning: foreach() argument must be of type array|object, null given in grants_handler_webform_element_alter() (line 244 of modules/custom/grants_handler/grants_handler.module).

**Comment by jiisuominen** [comment]
Äh, olipa tyhmä virhe. Unohin tehä tsekin uudessa hakemuksessa. Jännä etten ite huomannu aikasemmin. No nyt on fiksattu.

**Review by strangeworlder — APPROVED** [review]

File: public/modules/custom/grants_handler/grants_handler.module [hunk 1]
```diff
@@ -289,12 +357,12 @@ function grants_handler_webform_submission_form_alter(&$form, FormStateInterface
     // pass variable to Drupal.Settings
     $form['#attached']['drupalSettings']['grants_handler']['grantsProfile'] = $grantsProfile;
 
-    $bankAccountOptions = [];
-    foreach ($grantsProfile['bankAccounts'] as $key => $account) {
-      if (isset($account['confirmationFile'])) {
-        $bankAccountOptions[$account['bankAccount']] = $account['bankAccount'];
-      }
-    }
+    //    $bankAccountOptions = [];
```
Comment (strangeworlder) [inline public/modules/custom/grants_handler/grants_handler.module:360]: Onko jotain, joka voidaan poistaa uloskommentoinnin sijaan?

File: public/modules/custom/grants_handler/grants_handler.module [hunk 2]
```diff
@@ -305,7 +373,8 @@ function grants_handler_webform_submission_form_alter(&$form, FormStateInterface
         t('Oma asiointi > Omat tiedot -osioon, jotta sitä voidaan käyttää tällä hakemuksella.'));
     }
 
-    $form["elements"]["1_hakijan_tiedot"]["tilinumero"]["account_number_select"]["#options"] = $bankAccountOptions;
+    //    $form["elements"]["1_hakijan_tiedot"]["tilinumero"]["account_number_select"]["#options"] = $bankAccountOptions;
```
Comment (strangeworlder) [inline public/modules/custom/grants_handler/grants_handler.module:376]: tämä siis toinen kohta, jossa uloskommentoitu

**Review by strangeworlder — APPROVED** [review]

**Pull request closed without merging**